pallets/interest-accrual/src/lib.rs

```diff
@@ -132,82 +160,70 @@ pub mod pallet {
 		DebtAdjustmentFailed,
 		/// Emits when the interest rate was not used
 		NoSuchRate,
+		/// Emits when a historic rate was asked for from the future
+		NotInPast,
 	}
 
-	// TODO: add permissionless extrinsic to update any rate
+	#[pallet::hooks]
+	impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {
+		fn on_initialize(_: T::BlockNumber) -> Weight {
+			let mut count = 0;
+			let then = LastUpdated::<T>::get();
+			let now = Self::now();
+			LastUpdated::<T>::set(Self::now());
+			let delta = Self::now() - then;
+			let bits = Moment::BITS - delta.leading_zeros();
+			Rate::<T>::translate(|per_sec, mut rate: RateDetailsOf<T>| {
+				count += 1;
+				Self::calculate_accumulated_rate(per_sec, rate.accumulated_rate, then, now)
+					.ok()
+					.map(|new_rate| {
+						rate.accumulated_rate = new_rate;
+						rate
+					})
+			});
+			T::DbWeight::get().reads_writes(2, 1)
+				+ count
+					* (T::DbWeight::get().reads_writes(1, 1)
+						+ T::Weights::calculate_accumulated_rate(bits))
+		}
+	}
```

Comment:
We should call `upgrade_to_v1` here or we should provide a costume migration struct that altair can use in order to upgrade. 

And if we have a `StorageVersion` we should provide a on genesis builder that uses the latest `StorageVersion`. Hence again, would strive for enum as `StorageVersion`.

Reply:
You're right, we do need to add that to the genesis. I'll take care of that with the rest of the updates